leader-election-impl/src/main/java/com/palantir/paxos/SingleLeaderLearnerNetworkClient.java

```diff
@@ -63,18 +64,23 @@ public SingleLeaderLearnerNetworkClient(
     public void learn(long seq, PaxosValue value) {
         // broadcast learned value
         for (final PaxosLearner learner : remoteLearners) {
-            executors.get(learner).execute(() -> {
-                try {
-                    learner.learn(seq, value);
-                } catch (Throwable e) {
-                    log.warn("Failed to teach learner the value {} at sequence {}",
-                            UnsafeArg.of("value", Optional.ofNullable(value.data)
-                                    .map(bytes -> BaseEncoding.base16().encode(bytes))
-                                    .orElse(null)),
-                            SafeArg.of("sequence", seq),
-                            e);
-                }
-            });
+            try {
+                executors.get(learner).execute(() -> {
+                    try {
+                        learner.learn(seq, value);
+                    } catch (Throwable e) {
+                        log.warn("Failed to teach learner the value {} at sequence {}, after attempting execution.",
+                                UnsafeArg.of("value", base16EncodePaxosValue(value)),
+                                SafeArg.of("sequence", seq),
+                                e);
+                    }
+                });
+            } catch (RejectedExecutionException e) {
+                log.warn("Failed to teach learner the value {} at sequence {}, because we could not execute the task.",
```

Comment:
As below: this happens if the executor has the max number of threads running (e.g. because the calls to that endpoint are slow).

Reply:
as an aside, what are your thoughts on making the logging message the same across all instances of this particular exception (as far as this PR goes, at least). It would make analysing issues easier - you could just look for log lines with the same exact message.